src/hooks/useLoginCallbackCheck.tsx

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import * as React from 'react';
+import { AuthSdkError, OktaAuth } from '@okta/okta-auth-js';
+
+const LOGIN_CALLBACK_TIMEOUT = 1000;
+
+const useLoginCallbackCheck = (
+  oktaAuth: OktaAuth
+): {
+  loginCallbackError?: AuthSdkError
+} => {
+  const [loginCallbackError, setLoginCallbackError] = React.useState<AuthSdkError | undefined>(undefined);
+
+  React.useEffect(() => {
+    let loginCallbackTimer: NodeJS.Timeout;
+    if (oktaAuth?.isLoginRedirect()) {
+      loginCallbackTimer = setTimeout(() => {
```

Comment:
I'm not sure I fully understand what this hook does, and the usage of `setTimeout` makes me nervous of race conditions and timing issues

Reply:
I think this `useLoginCallbackCheck` file can be deleted.
I tried to add some helper for developers to detect incorrect routes configuration when login callback is bound to a wrong route or no route.